xbmc/GUIPassword.cpp

```diff
@@ -117,6 +117,79 @@ bool CGUIPassword::IsItemUnlocked(CFileItem* pItem, const std::string &strType)
   return true;
 }
 
+bool CGUIPassword::IsItemUnlocked(CMediaSource* pItem, const std::string &strType)
+{
+  const std::shared_ptr<CProfileManager> profileManager = CServiceBroker::GetSettingsComponent()->GetProfileManager();
+
+  // \brief Tests if the user is allowed to access the MediaSource
+  // \param pItem The share folder item to access
+  // \param strType The type of share being accessed, e.g. "music", "video", etc. See CSettings::UpdateSources()
+  // \return If access is granted, returns \e true
+  if (profileManager->GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE)
+    return true;
+
+  while (pItem->m_iHasLock > 1)
+  {
+    std::string strLockCode = pItem->m_strLockCode;
+    std::string strLabel = pItem->strName;
+    int iResult = 0;  // init to user succeeded state, doing this to optimize switch statement below
+    char buffer[33]; // holds 32 places plus sign character
+    if(g_passwordManager.bMasterUser)// Check if we are the MasterUser!
```

Comment:
Why not just do if (!g_passwordManager... As you already set iresult to 0 anyway

Reply:
cause it's a copy of previously existing IsItemUnlocked(**CFileItem*** pItem... i left it as is and changed the two little hunks. IsItemUnlocked(**CMediaSource*** pItem... wasn't implemented before...it was just defined in the header.

Reply:
It's fine to reuse, but if you are going to, might as well make it better while you are at it

Reply:
fine, then i'll throw them both out

Reply:
**the answer**
/home/fhw/xbmc/xbmc/GUIPassword.cpp: In member function ‘bool CGUIPassword::IsItemUnlocked(CFileItem*, const string&)’:
/home/fhw/xbmc/xbmc/GUIPassword.cpp:62:7: error: ‘iResult’ was not declared in this scope
       iResult = 0;
       ^~~~~~~

never touch a running system :)

Reply:
I think you misunderstood me. Instead of if () iresult = 0 } else {}, just change it to if (!g_pasdword...) { Stuff in your else}, all you do in the if is set a var you have already set, so skip that, and check for the negation

Reply:
thanks, done

Reply:
thanks, done